### PR TITLE
ncm-network: add IPv6 support

### PR DESCRIPTION
**Pull request by stdweird**
Also includes some code cleanup and more unittests

### CODE REVIEW

**Comment by stdweird** [comment]
This is still WIP and requires #423 to be merged in

**Comment by jrha** [comment]
I'm guessing this needs rebasing now that #423 is merged.

File: ncm-network/src/main/perl/network.pm [hunk 1]
```diff
@@ -316,7 +282,7 @@ sub Configure {
                     $text .= "OVS_EXTRA='$net{$iface}{ovs_extra}'\n";
                 }
                 if ($net{$iface}{'bond_ifaces'}) {
-                    $text .= "BOND_IFACES=".join(' ',$net{$iface}{bond_ifaces})."\n";
+                    $text .= "BOND_IFACES='".join(' ',@{$net{$iface}{bond_ifaces}})."'\n";
```
Comment (Pansanel) [inline ncm-network/src/main/perl/network.pm]: You are right. The lack of quotes was an issue.

**Comment by stdweird** [comment]
@jrha rebased

**Comment by stdweird** [comment]
we are running this branch without any ipv6 bits, so all old behaviour should still be intact
@dvandok reported that they are running a modified version with dual stack.

File: ncm-network/src/main/perl/network.pm [hunk 2]
```diff
@@ -609,51 +636,76 @@ sub Configure {
...
+            my $if = $1;
+            # ifdown: all devices that have files with non-zero status
             if ($exifiles{$file} != 0) {
-                $self->debug(3,"exifiles file $file with non-zero value found: ".$exifiles{$file});
-                $ifdown{$if}=1;
-                ## bonding: if you bring down a slave, allways bring
-                ## down it's master
+                $self->debug(3, "exifiles file $file with non-zero value found: ".$exifiles{$file});
+                $ifdown{$if} = 1;
+                # bonding: if you bring down a slave, allways bring
```
Comment (jrha) [inline ncm-network/src/main/perl/network.pm]: always

**Comment by jrha** [comment]
Do the type definitions really belong in the component or at a higher level?

**Comment by stdweird** [comment]
@jrha i don't know. some of them are very redhat/initscripts specific tough. and i'm unaware of any other code that reuses the types.

**Comment by jrha** [comment]
Thanks, I'm happy with this, will leave it a bit longer in case anyone wants to look it over.

**Pull request merged**